From 36d4bd91e686365009ded9c4925f17b0203cc2cb Mon Sep 17 00:00:00 2001
From: Caspar Friedrich <c.s.w.friedrich@gmail.com>
Date: Sun, 1 Mar 2020 18:36:22 +0100
Subject: [PATCH] Add functions to check in which state the stack actually is.

Signed-off-by: Caspar Friedrich <c.s.w.friedrich@gmail.com>
---
 include/esb.h    | 12 ++++++++++++
 subsys/esb/esb.c | 13 +++++++++++++
 2 files changed, 25 insertions(+)

diff --git a/include/esb.h b/include/esb.h
index c8853b42..e798ce50 100644
--- a/include/esb.h
+++ b/include/esb.h
@@ -276,6 +276,18 @@ void esb_disable(void);
  */
 bool esb_is_idle(void);
 
+/** @brief Check if the Enhanced ShockBurst module is in rx mode.
+ *
+ *  @return True if the module is in rx mode, false otherwise.
+ */
+bool esb_is_rx(void);
+
+/** @brief Check if the Enhanced ShockBurst module is in tx mode.
+ *
+ *  @return True if the module is in tx mode, false otherwise.
+ */
+bool esb_is_tx(void);
+
 /** @brief Write a payload for transmission or acknowledgement.
  *
  *  This function writes a payload that is added to the queue. When the module
diff --git a/subsys/esb/esb.c b/subsys/esb/esb.c
index 59b9ba21..049ae2d6 100644
--- a/subsys/esb/esb.c
+++ b/subsys/esb/esb.c
@@ -1140,6 +1140,19 @@ bool esb_is_idle(void)
 	return (esb_state == ESB_STATE_IDLE);
 }
 
+bool esb_is_rx(void)
+{
+	return (esb_state == ESB_STATE_PRX ||
+		esb_state == ESB_STATE_PRX_SEND_ACK);
+}
+
+bool esb_is_tx(void)
+{
+	return (esb_state == ESB_STATE_PTX_TX ||
+		esb_state == ESB_STATE_PTX_RX_ACK ||
+		esb_state == ESB_STATE_PTX_TX_ACK);
+}
+
 int esb_write_payload(const struct esb_payload *payload)
 {
 	if (!esb_initialized) {
-- 
2.25.1

